bve-obj-conv/src/main.rs

```diff
@@ -0,0 +1,236 @@
+use async_std::task::block_on;
+use bve::filesystem::read_convert_utf8;
+use itertools::Itertools;
+use obj::{Obj, SimplePolygon};
+use std::{
+    collections::HashMap,
+    fs::File,
+    io::{BufRead, Cursor, Write},
+    panic::catch_unwind,
+    path::PathBuf,
+};
+
+const HEADER: &str = concat!(
+    r#"BVE-Reborn .obj to .csv Converter
+https://github.com/bve-reborn/bve-reborn
+Version "#,
+    env!("CARGO_PKG_VERSION"),
+    "\n\n"
+);
+
+fn print(data: impl AsRef<[u8]>) {
+    let stdout = std::io::stdout();
+    let mut lock = stdout.lock();
+    lock.write_all(data.as_ref()).expect("Could not write to stdout");
+    lock.flush().expect("Could not flush stdout");
+}
+
+fn read_data() -> String {
+    let stdin = std::io::stdin();
+    let mut lock = stdin.lock();
+    let mut string = String::new();
+    lock.read_line(&mut string).expect("Could not read input");
+    string.pop(); // remove newline
+    if let Some(idx) = string.find('"') {
+        string = string.chars().skip(idx + 1).take_while(|c| *c != '"').collect();
+    };
+    string
+}
+
+enum State {
+    Input,
+    Wrong,
+}
+
+fn get_input<T>(prompt: impl AsRef<[u8]>, mut verification: impl FnMut(Option<PathBuf>) -> Result<T, String>) -> T {
+    let mut state = State::Input;
+    loop {
```

Comment:
I don't like this

The structure of this function just feels wrong to me. I guess you want to validate input? But you have a lot of steps between "receive input from the user" and "do something with the input"

Reply:
I mean I don't know how else you'd want me to split this up, it receives input, pre-processes it into a usable form, then verifies it with the provided function.